scripts/eslint-plugin/css_before_spread_props.js

```diff
@@ -0,0 +1,47 @@
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Enforce CSS props are declared before spread props',
+    },
+  },
+  create: function(context) {
+    const allElementsArr = [];
+
+    return {
+      JSXElement(node) {
+        const tempArr = [];
+        node.openingElement.attributes.forEach(attribute => tempArr.push(attribute));
+        allElementsArr.push(tempArr);
+      },
+      'Program:exit'() {
+        allElementsArr.forEach(elementArr => {
+          const regex = new RegExp('^(\{\.\.\.).+\}$');
+          const cssPropsIndex = elementArr.findIndex(node => (node.name && node.name.name === 'css'));
+          const spreadPropsIndex = elementArr.findIndex(node => (node.type === 'JSXSpreadAttribute'));
+
+          if (cssPropsIndex === -1) {
+            return;
+          }
+
+          if (spreadPropsIndex === -1) {
+            return;
+          }
```

Comment:
[optional nit/microperf] I'd rewrite this with even earlier returns if possible - that way if a `css` prop doesn't exist, there's no need to look for a spread/rest at all
```suggestion
          const cssPropsIndex = elementArr.findIndex(node => (node.name && node.name.name === 'css'));
          if (cssPropsIndex === -1) return;

          const spreadPropsIndex = elementArr.findIndex(node => (node.type === 'JSXSpreadAttribute'));
          if (spreadPropsIndex === -1) return;
```

i also slightly prefer single line returns for early returns, but that's a personal idiosyncrasy